minimum_jerk_ros/src/minimum_jerk_ros.cpp

```diff
@@ -0,0 +1,736 @@
+        {
+            FilFile::close_file(fd_for_times);
+            FilFile::close_file(fd_r_s_times);
+        }
+        try
+        {
+            this->current_pose_ = this->get_current_pose_();
+        }
+        catch (tf2::TransformException &e)
+        {
+            action_rotation_server_->terminate_current();
+            return;
+        }
+
+        rotation_feedback->angular_distance_traveled = compute_angle_();
+        auto res = std::make_shared<Rotation::Result>();
+        res->total_elapsed_time.nanosec = int((get_clock()->now() - t_init).nanoseconds());
+        RCLCPP_INFO(this->get_logger(), "distance traveled %f, error %f\n", rotation_feedback->angular_distance_traveled, abs(rotation_feedback->angular_distance_traveled - abs(goal->target_yaw)));
+        (abs(rotation_feedback->angular_distance_traveled - abs(goal->target_yaw)) <= yaw_tolerance) ? action_rotation_server_->succeeded_current(res) : action_rotation_server_->terminate_current();
+    }
+
+    void MinimumJerkRos::translation_callback()
+    {
+        auto goal = action_translation_server_->get_current_goal();
+        RCLCPP_INFO(this->get_logger(), "Translation: %f m, min_velocity: %f, collision_check: %i, xy_goal_tolerance: %f, data_save: %i",
+                    goal->target_x, goal->min_velocity, goal->enable_collision_check, goal->xy_goal_tolerance, goal->enable_data_save);
+        auto t_init = get_clock()->now();
+        try
+        {
+            this->init_pose_ = this->get_current_pose_();
+        }
+        catch (tf2::TransformException &e)
+        {
+            action_translation_server_->terminate_current();
+            return;
+        }
+        auto translation_feedback = std::make_shared<Translation::Feedback>();
+        auto linear_vel = geometry_msgs::msg::Twist();
+        linear_vel.linear = geometry_msgs::msg::Vector3();
+        double xy_tolerance = 0.015;
+        if (goal->xy_goal_tolerance > 0)
+            xy_tolerance = goal->xy_goal_tolerance;
+        auto odometry = compute_translation_velocities_(goal);
+
+        auto poses = convert_poses_to_posestampeds_(odometry.get_poses());
+        if (poses.size() == 0)
+        {
+            RCLCPP_ERROR(get_logger(), "Error transformation of the path");
+            action_translation_server_->terminate_all();
+            return;
+        }
+
+        if (debug_trajectory_)
+        {
+            show_trajectory_(poses);
+        }
+
+        int i = 0;
+
+        int fd_for_times;
+        int fd_r_s_times;
+        if (goal->enable_data_save)
+        {
+            fd_for_times = FilFile::open_file("./src/mypackage/minimum_jerk_relative_move/minimum_jerk_trajectory_planner/ressources/for_times.txt");
+            fd_r_s_times = FilFile::open_file("./src/mypackage/minimum_jerk_relative_move/minimum_jerk_trajectory_planner/ressources/r_s_times.txt");
+        }
+        auto t_init_for = get_clock()->now();
+        for (auto vel : odometry.get_velocities())
```

Comment:
what happens if a second goal is sent before the end of this one (i.e. `action_translation_server_->is_preempt_requested()` returns true), does it disrupts the behavior ? 
ps: not saying you should do anything fancy about it, imo it shouldn't disrupting the behavior and you should discard preemption